Robot2018/src/org/usfirst/frc/team199/Robot2018/OI.java

```diff
@@ -34,4 +46,27 @@
 	// Start the command when the button is released and let it run the command
 	// until it is finished as determined by it's isFinished method.
 	// button.whenReleased(new ExampleCommand());
+	
+	public Joystick leftJoy;
+		public JoystickButton shiftDrive;
+		public JoystickButton shiftDriveType;
+	public Joystick rightJoy;
+	public Joystick manipulator;
+	
+	
+	public int getButton(String key, int def) {
```

Comment:
The joysticks need to be public because their values are needed in some functions. I have no clue about the buttons, however, that's how we did it last year so that's how I did it this week.